Core/include/Acts/Vertexing/TrackAtVertex.hpp

```diff
@@ -74,4 +74,24 @@ struct TrackAtVertex {
   bool isLinearized = false;
 };
 
+struct TrackAtVertexRef {
```

Comment:
Now this is a bit of a stopgap solution to replacing track at vertex completely with a type erasure type. This autoconstructs from a track at vertex, and maintains references.

Reply:
I don't quite understand why this is needed, could you explain?